src/contentMetaFilter.js

```diff
@@ -0,0 +1,73 @@
+var path = require('path')
+var filename = path.basename(__filename)
+var _ = require('lodash')
+
+// Function to generate the Config Array
+function generateConfigString (metaFiltersArray) {
+  var configArray = {}
+  _.forOwn(metaFiltersArray, function (value, key) {
+    const allowedMetadata = value[0]
+    const blackListedMetadata = value[1]
+    if ((allowedMetadata && allowedMetadata.length > 0) && (blackListedMetadata && blackListedMetadata.length > 0)) {
+      configArray[key] = _.difference(allowedMetadata, blackListedMetadata)
+    } else if (allowedMetadata && allowedMetadata.length > 0) {
+      configArray[key] = allowedMetadata
+    } else if (blackListedMetadata && blackListedMetadata.length > 0) {
+      configArray[key] = { 'ne': blackListedMetadata }
+    }
+  })
+  LOG.info(utilsService.getLoggerData({}, 'INFO',
+    filename, 'getFilterConfig', 'config array', configArray))
+  return configArray
+}
+// function to generate the search filter and return JSON Object
+function getMetaFilterConfig () {
+  LOG.info(utilsService.getLoggerData({}, 'INFO',
+    filename, 'getFilterConfig', 'environment info', process.env))
+  var allowedChannels = process.env.sunbird_content_service_whitelisted_channels
+    ? process.env.sunbird_content_service_whitelisted_channels.split(',') : []
+  var blackListedChannels = process.env.sunbird_content_service_blacklisted_channels
+    ? process.env.sunbird_content_service_blacklisted_channels.split(',') : []
+  var allowedFramework = process.env.sunbird_content_service_whitelisted_framework
+    ? process.env.sunbird_content_service_whitelisted_framework.split(',') : []
+  var blackListedFramework = process.env.sunbird_content_service_blacklisted_framework
+    ? process.env.sunbird_content_service_blacklisted_framework.split(',') : []
+  var allowedMimetype = process.env.sunbird_content_service_whitelisted_mimetype
+    ? process.env.sunbird_content_service_whitelisted_mimetype.split(',') : []
+  var blackListedMimetype = process.env.sunbird_content_service_blacklisted_mimetype
+    ? process.env.sunbird_content_service_blacklisted_mimetype.split(',') : []
+  var allowedContenttype = process.env.sunbird_content_service_whitelisted_contenttype
+    ? process.env.sunbird_content_service_whitelisted_contenttype.split(',') : []
+  var blackListedContenttype = process.env.sunbird_content_service_blacklisted_contenttype
+    ? process.env.sunbird_content_service_blacklisted_contenttype.split(',') : []
+  var allowedResourcetype = process.env.sunbird_content_service_whitelisted_resourcetype
+    ? process.env.sunbird_content_service_whitelisted_resourcetype.split(',') : []
+  var blackListedResourcetype = process.env.sunbird_content_service_blacklisted_resourcetype
+    ? process.env.sunbird_content_service_blacklisted_resourcetype.split(',') : []
+
+  // Check if the Filter Config service data is defined, if yes, create Object with it
+  const filterConfigService = ''
```

Comment:
TODO please don't make an empty string and use the empty string check,
make the filterConfigService as null and do the null check

Reply:
Is this comment addressed?